projects/rocprofiler-sdk/source/lib/rocprofiler-sdk/pc_sampling/ioctl/ioctl_adapter.cpp

```diff
@@ -525,7 +580,9 @@ ioctl_query_pcs_configs(const rocprofiler_agent_t* agent, rocp_pcs_cfgs_vec_t& r
             // This should never happened, unless the KFD is broken.
             continue;
         }
-        rocp_configs.emplace_back(rocp_cfg);
+        if(check_firmware_compatibility(agent, get_rocp_pcs_method_from_kfd(ioctl_cfg.method)) ==
+           ROCPROFILER_STATUS_SUCCESS)
+            rocp_configs.emplace_back(rocp_cfg);
```

Comment:
Maybe just a preference in case we decided to introduce some more checks in the future. Then we could avoid further nesting the code.
```suggestion
        if(check_firmware_compatibility(agent, get_rocp_pcs_method_from_kfd(ioctl_cfg.method)) !=
           ROCPROFILER_STATUS_SUCCESS)
           continue;
        rocp_configs.emplace_back(rocp_cfg);
```